### PR TITLE
Password-protect private key reveal

### PR DESCRIPTION
**Pull request by yrliou**
Resolves https://github.com/brave/brave-browser/issues/24830

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Unlock the Brave wallet
2. Navigate to Accounts -> Select Account1 -> Account details -> Private Key
3. "Show key" button should be disabled & and password field should be visible
4. Enter wallet password into the password field
5. "Show key" button should now be enabled
6a. Error message appears if incorrect password was entered
6b. Key is displayed if correct password was entered

Screen Shot 2022-07-27 at 6 08 02 PM
Screen Shot 2022-07-27 at 6 08 13 PM
Screen Shot 2022-07-27 at 6 09 59 PM
Screen Shot 2022-07-27 at 6 12 50 PM

### CODE REVIEW

**Comment by yrliou** [comment]
cc @darkdh to review the backend commit

**Review by darkdh — APPROVED** [review]
backend++

**Comment by SergeyZhukovsky** [comment]
@Pavneet-Sing would you mind to review Android changes for that?

File: android/java/org/chromium/chrome/browser/crypto_wallet/model/OnboardingViewModel.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+
+public class OnboardingViewModel extends ViewModel {
+    private final MutableLiveData<String> mPassword = new MutableLiveData<>();
+
+    public void setPassword(String password) {
+        mPassword.setValue(password);
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/crypto_wallet/model/OnboardingViewModel.java:16]: nit: might wanna use `postValue` for thread safety, currently it's always being called on main thread so no issue and could be intentional, feel free to resolve.

File: android/java/org/chromium/chrome/browser/crypto_wallet/fragments/onboarding_fragments/RecoveryPhraseFragment.java [hunk 2]
```diff
@@ -56,21 +60,28 @@ public View onCreateView(
...
+                    recoveryPhrases = Utils.getRecoveryPhraseAsList(result);
+                    setupRecoveryPhraseRecyclerView(view);
+                    TextView copyButton = view.findViewById(R.id.btn_copy);
+                    assert getActivity() != null;
+                    copyButton.setOnClickListener(v -> {
+                        Utils.saveTextToClipboard(getActivity(),
+                                Utils.getRecoveryPhraseFromList(recoveryPhrases),
+                                R.string.text_has_been_copied, true);
+                    });
+                    setupRecoveryPhraseRecyclerView(view);
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/crypto_wallet/fragments/onboarding_fragments/RecoveryPhraseFragment.java:81]: nit: `setupRecoveryPhraseRecyclerView(view)` is called twice

**Review by Pavneet-Sing — APPROVED** [review]
LGTM

File: components/brave_wallet_ui/page/async/wallet_page_async_handler.ts [hunk 3]
```diff
@@ -194,7 +194,8 @@ handler.on(WalletPageActions.removeImportedAccount.getType(), async (
 
 handler.on(WalletPageActions.viewPrivateKey.getType(), async (store: Store, payload: ViewPrivateKeyPayloadType) => {
   const keyringService = getWalletPageApiProxy().keyringService
-  const result = await keyringService.getPrivateKeyForKeyringAccount(payload.address, payload.coin)
+  // TODO (JL): Pass password here.
```
Comment (yrliou) [inline components/brave_wallet_ui/page/async/wallet_page_async_handler.ts:197]: cc @josheleonard ^^

Comment (josheleonard) [reply components/brave_wallet_ui/page/async/wallet_page_async_handler.ts:197]: Need this to get merged first: https://github.com/brave/brave-core/pull/14669

Comment (josheleonard) [reply components/brave_wallet_ui/page/async/wallet_page_async_handler.ts]: now passing password to `keyringService.getPrivateKeyForKeyringAccount` calls

[fix: use api-proxy & password to manage key reveal](https://github.com/brave/brave-core/pull/14731/commits/6507da55b03b5725df886ec0d71fa1b16b21d3b3)

**Review by muliswilliam — APPROVED** [review]

**Review by kdenhartog — APPROVED** [review]
Looks good to me! thanks for finishing this one up

**Pull request merged**